packages/hardhat/scripts/new-dao.ts

```diff
@@ -0,0 +1,220 @@
+    stakingFactory,
+    feeToken,
+    challangeDuration,
+    actionAmount,
+    challangeAmount
+  } = transform(await import(`../params-${network}.json`));
+
+  console.log(transform(await import(`../params-${network}.json`)))
+
+  const createDaoTxOne = async (gardensTemplate: GardensTemplate, log: Function): Promise<string> => {
+    const tx = await gardensTemplate.createDaoTxOne(
+      orgTokenName,
+      orgTokenSymbol,
+      fundingPoolStake,
+      [
+        voteDuration,
+        voteSupportRequired,
+        voteMinAcceptanceQuorum,
+        delegatedVotingPeriod,
+        voteQuietEndingPeriod,
+        voteQuietEndingExtension,
+        voteExecutionDelay
+      ],
+      { gasLimit: 9500000 }
+    );
+
+    const daoAddress = await getAddress("DeployDao", gardensTemplate, tx.hash);
+
+    log(`Tx one completed: Gardens DAO (${daoAddress}) created.`);
+
+    return daoAddress
+  };
+
+  const createTokenholders = async (gardensTemplate: GardensTemplate, log: Function): Promise<void> => {
+    const tx = await gardensTemplate.createTokenholders(
+      holders,
+      stakes,
+      { gasLimit: 9500000 }
+    )
+
+    log(`Tx tokenholders completed.`);
+  }
+
+  const createDaoTxTwo = async (gardensTemplate: GardensTemplate, log: Function): Promise<void> => {
+
+    const tx = await gardensTemplate.createDaoTxTwo(
+      [issuanceTargetRatio, issuanceMaxAdjustmentPerSecond],
+      stableTokenAddress,
+      stableTokenOracle,
+      [decay, maxRatio, weight, minEffectiveSupply],
+      { gasLimit: 9500000 }
+    );
+
+    log(`Tx two completed.`);
+  };
+
+  const createDaoTxThree = async (gardensTemplate: GardensTemplate, log: Function): Promise<void> => {
+    const tx = await gardensTemplate.createDaoTxThree(
+      arbitrator,
+      setAppFeesCashier,
+      agreementTitle,
+      ethers.utils.toUtf8Bytes(agreementContent),
+      stakingFactory,
+      feeToken,
+      challangeDuration,
+      [actionAmount, challangeAmount],
+      { gasLimit: 9500000 }
+    );
+
+    log(`Tx three completed.`);
+  };
+
+  const appManager = await ethers.getSigners()[0];
+  const gardensTemplate = await getGardensTemplate(appManager)
+
+  const daoAddress = await createDaoTxOne(gardensTemplate, log);
+  await createTokenholders(gardensTemplate, log)
+  await createDaoTxTwo(gardensTemplate, log);
+  await createDaoTxThree(gardensTemplate, log);
+
+  const [convictionVotingAddress, tokenManagerAddress, issuanceAddress, agreementAddress, votingAddress] = await getApps(
+    daoAddress,
+    await Promise.all([
+      gardensTemplate.CONVICTION_VOTING_APP_ID(),
+      gardensTemplate.HOOKED_TOKEN_MANAGER_APP_ID(),
+      gardensTemplate.DYNAMIC_ISSUANCE_APP_ID(),
+      gardensTemplate.AGREEMENT_APP_ID(),
+      gardensTemplate.DISPUTABLE_VOTING_APP_ID(),
+    ])
+  )
+
+  return {
+    daoAddress,
+    convictionVotingAddress,
+    tokenManagerAddress,
+    issuanceAddress,
+    agreementAddress,
+    votingAddress,
+  }
```

Comment:
AFAIK none of this is used, can it be removed?

Reply:
The `main()` function can be used by other files to deploy a DAO, for example for tests or mock actions, so I would like to keep the return statement. In the last commit I also added a log so we can recover those addresses in the console as well.